server/src/main/java/org/hl7/davinci/endpoint/rems/controller/RemsController.java

```diff
@@ -119,14 +119,21 @@ public ResponseEntity<Object> postRems(@RequestBody String jsonData) {
         remsRequest.setResource(remsObject);
         remsRepository.save(remsRequest);
 
-        // this loop needs to change to handle multiple levels of sub-requirement conditions
-        // this loop needs to also handle parsing out resources for each requirement - may need to be separate endpoints
         for (Requirement requirement : drug.getRequirements()) {
             MetRequirement metReq = new MetRequirement();
             metReq.setRequirement(requirement);
             metReq.setRemsRequest(remsRequest);
             remsRequest.addMetRequirement(metReq);
             metRequirementsRepository.save(metReq);
+
+            for (Requirement subRequirement : requirement.getChildRequirements()) {
+              MetRequirement subMetReq = new MetRequirement();
+              subMetReq.setRequirement(subRequirement);
+              subMetReq.setRemsRequest(remsRequest);
```

Comment:
only support one level of sub requirements - this would need to change to a dfs/bfs implementation for multiple levels of nested sub requirements